### ISSUE TITLE
Access to phone address book

### ISSUE THREAD

**Issue by fyahfox** [NewEconomyMovement/NEMAndroidApp#21]
A lot of people dislike it a lot if apps have access to the phone address book and at the same time internet access, because that could mean that all contacts are sent somewhere.

I would really like to discuss this again and strongly suggest to make a separate address book for the NEM app without access to the phone address book. The advantages of an access do not outweigh the disadvantages (privacy issue) imo.

**Comment by kmiliak** [comment]
App hasn't access to address book unless user allow it in settings.
From our part, we can do separate address book for NEM, it won't take too much time. But without importing accounts from phone address and vice versa ('cause this will take more time).
Also, we cannot implement this now, as developers have their primary issues, and other additional functionalities, improvements will be done after we finish with current version. As it is hard for developers to switch between tasks.

**Comment by fyahfox** [comment]
Yes, it was more meant as question/discussion anyway.

I never saw any settings for this in the Android version?

**Comment by kmiliak** [comment]
I meant that when user installs app, user sees the message that app requires access, so in this way they are notified. 
Anyway, we can make separate address book for both versions, if you decide. Just let us know requirements for NEM address book.

**Comment by fyahfox** [comment]
They are notified, yes... But only since Android 6 users can decide to install the app without phone contacts access. Android 5 and lower users can just decide to not install the app at all.

There are pros and cons for both ways, so we should discuss that.

**Comment by am4n0w4r** [comment]
I tried to think about actual benefits of storing contacts in phone book and came up with only two things:
Contacts are not deleted when app is uninstalled and they usually are synchronized with google servers if device owner allows it. So if something happens with app/device you won't lose your contacts. Although in this situation I'd rather worry about my account keys first... That's all I could came up with for now.

**Comment by fyahfox** [comment]
Same here. So I think a better solution is a separated address book with an easy one click backup feature that will backup your accounts + addresses.

But maybe other people have more ideas.

**Comment by fyahfox** [comment]
I still think that the best solution is a one click backup feature inside the app that will backup/export accounts + address book, but this will be postponed for now.

Until then the contacts should be stored in the phone address book like it is now.

I will leave this issue open for future discusion.

**Comment by zachamo** [comment]
I fully agree with Fyahfox on the need to have a totally separate set of contacts.. If anything, there should be an option to add a NEM contact to your main address book (i.e. when editing contacts, a button "Add to Device Contacts", or a setting where you can default synchronization across the two.. I think it should be assumed that they are not synchronized. You can default synchronization in 10 years if NEM goes mainstream, for now it's a niche app that doesn't seamlessly integrate with people's daily lives. 

Keeping NEM address seperate also means your addresses are behind your NEM password, and not sitting in your google address book. As a user, I may not want this information to be so broadly available to Google / people snooping my device.. What if I pay for my Ashley Madison account with NEM -- don't want that in my google contacts for my wife to find! :laughing:
